tests/v2/validation/upgrade/workload.go

```diff
@@ -31,26 +37,339 @@ import (
+	}
+
+	logrus.Infof("Checking daemonsets in namespace %s", namespace.Name)
+	daemonsetList, err := steveClient.SteveType(workloads.DaemonsetSteveType).List(nil)
+	require.NoError(t, err)
+	daemonsetNames := []string{
+		names.coreWithSuffix[daemonsetName],
+	}
+
+	for _, expectedDaemonsetName := range daemonsetNames {
+		doesContainDaemonset := containsItemWithPrefix(daemonsetList.Names(), expectedDaemonsetName)
+		assert.Truef(t, doesContainDaemonset, "Daemonset with prefix %s doesn't exist", expectedDaemonsetName)
+	}
+
+	if *featuresToTest.Ingress {
+		logrus.Infof("Checking deployment for ingress in namespace %s", namespace.Name)
+		doesContainDeploymentForIngress := containsItemWithPrefix(deploymentList.Names(), names.coreWithSuffix[deploymentNameForIngress])
+		assert.Truef(t, doesContainDeploymentForIngress, "Deployment with prefix %s doesn't exist", names.coreWithSuffix[deploymentNameForIngress])
+
+		logrus.Infof("Checking daemonset for ingress in namespace %s", namespace.Name)
+		doesContainDaemonsetForIngress := containsItemWithPrefix(daemonsetList.Names(), names.coreWithSuffix[daemonsetNameForIngress])
+		assert.Truef(t, doesContainDaemonsetForIngress, "Daemonset with prefix %s doesn't exist", names.coreWithSuffix[daemonsetNameForIngress])
+
+		logrus.Infof("Checking ingresses in namespace %s", namespace.Name)
+		ingressList, err := steveClient.SteveType(ingresses.IngressSteveType).List(nil)
+		require.NoError(t, err)
+		ingressNames := []string{
+			names.coreWithSuffix[ingressNameForDeployment],
+			names.coreWithSuffix[ingressNameForDaemonset],
+		}
+
+		for _, expectedIngressName := range ingressNames {
+			doesContainIngress := containsItemWithPrefix(ingressList.Names(), expectedIngressName)
+			assert.Truef(t, doesContainIngress, "Ingress with prefix %s doesn't exist", expectedIngressName)
+
+			if doesContainIngress {
+				ingressName := getItemWithPrefix(ingressList.Names(), expectedIngressName)
+				ingressID := getSteveID(namespace.Name, ingressName)
+				ingressResp, err := steveClient.SteveType(ingresses.IngressSteveType).ByID(ingressID)
+				require.NoError(t, err)
+
+				ingressSpec := &networkingv1.IngressSpec{}
+				err = v1.ConvertToK8sType(ingressResp.Spec, ingressSpec)
+				require.NoError(t, err)
+
+				logrus.Infof("Checking if the ingress %s is accessible", ingressResp.Name)
+				isIngressAcessible, err := waitUntilIngressIsAccessible(client, ingressSpec.Rules[0].Host)
+				require.NoError(t, err)
+				assert.True(t, isIngressAcessible)
+			}
+		}
+	}
+
+	logrus.Infof("Checking the secret in namespace %s", namespace.Name)
+	secretList, err := steveClient.SteveType(secrets.SecretSteveType).List(nil)
+	require.NoError(t, err)
+
+	doesContainSecret := containsItemWithPrefix(secretList.Names(), names.core[secretName])
+	assert.Truef(t, doesContainSecret, "Secret with prefix %s doesn't exist", names.core[secretName])
+
+	if *featuresToTest.Chart {
```

Comment:
have you done any testing around the charts?

Reply:
I have not. I'll set that value to true in my next round of freeform jobs.